### PR TITLE
added a new script to add new codec to existing lhotse shars

### PR DESCRIPTION
**Pull request by paarthneekhara**
Should run in on multiple GPUs/nodes

I want to know if the work flow looks correct.

I tested training locally on shars generated by this script. Seems to be working ok.

### CODE REVIEW

File: scripts/magpietts/update_magpie_shars.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        if self._dataloader is None:
+            world_size, rank = ddp_info()
+            print("In model.setup - world size:", world_size, "rank:", rank)
+            cuts = CutSet.from_shar(in_dir=self.shar_root)
+            self.total_cuts = len(cuts)
+            self.approx_total_batches = self.total_cuts // (self.batch_size * world_size)
+            sampler = SimpleCutSampler(
+                cuts, shuffle=False, max_cuts=self.batch_size,
+                world_size=world_size, rank=rank,
+            )
```
Comment (XuesongYang) [inline scripts/magpietts/update_magpie_shars.py:131]: this may shard different items from the origin cuts shard. I observed this issue from my former attempts. it is worth to be double-checked. 

FYI, I implemented v2 version of lhotse shar prep, and avoided this issue. Will post a PR here once I got feedbacks from Piotr.

Brief intro: now the whole lhotse shar recipe was split into two steps: nemo manifest -> cuts/target_audio/context_audio shards -> extend with target_codes/context_codes.
Step 1: https://gitlab-master.nvidia.com/xueyang/nemo/-/blob/xueyang/magpie_release_2504-lhotse-v2-debug/scripts/magpietts/create_lhotse_shar_from_nemo_manifest.py?ref_type=heads
step 2: https://gitlab-master.nvidia.com/xueyang/nemo/-/blob/xueyang/magpie_release_2504-lhotse-v2-debug/scripts/magpietts/extend_lhotse_shards_with_audio_codes.py?ref_type=heads

File: scripts/magpietts/update_magpie_shars.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        return cut
+
+    def write_on_batch_end(self, trainer, pl_module, prediction, batch_indices, batch, batch_idx, dataloader_idx):
+        import time
+        st = time.time()
+        cuts_local, codes_local, context_codes_local = prediction["cuts"], prediction["codes"], prediction["context_codes"]
+        world_size = trainer.world_size
+        if world_size > 1:
+            gathered = [None] * trainer.world_size
+            torch.distributed.all_gather_object(gathered, (cuts_local, codes_local, context_codes_local))
```
Comment (XuesongYang) [inline scripts/magpietts/update_magpie_shars.py:65]: as suggested, all gather object would sacrifice the GPU utils.

File: scripts/magpietts/update_magpie_shars.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+def ddp_info():
+    if torch.distributed.is_available() and torch.distributed.is_initialized():
+        return torch.distributed.get_world_size(), torch.distributed.get_rank()
+    return 1, 0
+
+class CodecExtractor(pl.LightningModule):
+    def __init__(self, model_path, pad_multiple, batch_size, shar_root, num_workers):
+        super().__init__()
+        self.pad_multiple = pad_multiple
```
Comment (XuesongYang) [inline scripts/magpietts/update_magpie_shars.py]: we don't need this argument anymore since we can always get it from `self.codec_model.samples_per_frame`

**Comment by XuesongYang** [comment]
we need to resample to the target sample rate the same as codec models before running codec inference. Original audio shards stored the original sample rate. For example, libritts is 24kHz. It may or may not generate expected audio codes when using codec model with 22kHz.

File: scripts/magpietts/update_magpie_shars.py (hunk 1 above)
Comment (paarthneekhara) [reply scripts/magpietts/update_magpie_shars.py:131]: I think specifying max_cuts loads batches of same size sequentially and avoids this issue, but let me double check to confirm.

File: scripts/magpietts/update_magpie_shars.py (hunk 3 above)
Comment (paarthneekhara) [reply scripts/magpietts/update_magpie_shars.py]: That's right, changed this.

File: scripts/magpietts/update_magpie_shars.py (hunk 1 above)
Comment (paarthneekhara) [reply scripts/magpietts/update_magpie_shars.py:131]: I checked and the order of the cuts is preserved in the dataloader - Shuffle false and a fixed batch size should be the reason.

File: scripts/magpietts/update_magpie_shars.py (hunk 2 above)
Comment (paarthneekhara) [reply scripts/magpietts/update_magpie_shars.py:65]: I did not understand this suggestion - Should we do something else here?

**Comment by blisc** [comment]
Moved to https://github.com/NVIDIA/NeMo/pull/13548
